common/remote/hooks/azure/azurefile.cpp

```diff
@@ -1032,48 +1040,45 @@ class CAzureApiCopyClient : public CInterfaceOf<IAPICopyClient>
         }
         return false;
     }
-    virtual IAPICopyClientOp * startCopy(const char *srcPath, unsigned srcStripeNum,  const char *tgtPath, unsigned tgtStripeNum) override
+    virtual IAPICopyClientOp * startCopy(const char *srcPath, unsigned srcStripeNum,  const char *tgtPath, unsigned tgtStripeNum) const override
     {
-        if (!haveSASTokens)
-        {
-            sourceApiInfo->getSASToken(sourceSASToken);
-            targetApiInfo->getSASToken(targetSASToken);
-            haveSASTokens=true;
-        }
-        StringBuffer sourceURI, targetURI;
-        buildURI(sourceURI, sourceURIbase, srcStripeNum,  srcPath, sourceApiInfo, sourceSASToken);
-        buildURI(targetURI, targetURIbase, tgtStripeNum,  tgtPath, targetApiInfo, targetSASToken);
+        CAzureURI targetURI(tgtStripeNum,  tgtPath, targetApiInfo);
         Owned<IAPICopyClientOp> apiClient;
         if (isAzureFile(targetApiInfo->getStorageType()))
-            apiClient.setown(new AzureFileClient());
+            apiClient.setown(new AzureFileClient(targetURI.str()));
         else
-            apiClient.setown(new AzureBlobClient());
-        apiClient->startCopy(sourceURI, targetURI);
+            apiClient.setown(new AzureBlobClient(targetURI.str()));
```

Comment:
AzureBlobClient requires target URI to be a blob URI - azure file uri will cause an error.